### DISCUSSION TITLE
When using pipe, how to fail test when the stream writing process fails?

### DISCUSSION THREAD

**Discussion by velle** [bats-core/bats-core#952] [Q&A]
I have rewritten my test a bit, as below, to focus on the essentials for this question. 

@test "mytest" {
    cat \
        sql/load_spatialite.sql \
        sql/gpkgCreateBaseTables.sql \
        sql/gpkgInsertEpsgSRID25832.sql \
        | sqlite3 test.gpkg
    ./validate_gpkg.py test.gpkg
}

If one of the files do not exist, then cat exits with exit code 1. But because I pipe stdout to another process, the exit status of that command is 0. So the test passes even though a file was not found. 

Is there anything one can do to make the test fail, if the the pipe writing process (in this case `cat`) fails?

**Comment by martin-schulze-vireso** [comment]
Try `set -o pipefail`

**Reply by velle** [reply]
Perfect, thanks :)

**Comment by velle** [comment]
closing
